### PR TITLE
Add physical operator debugging

### PR DESCRIPTION
**Pull request by pstutz**
This PR allows to investigate how a CAPS physical operator gets translated to a Spark plan.

### CODE REVIEW

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/DataFrameOps.scala [hunk 1]
```diff
@@ -206,6 +211,46 @@ object DataFrameOps {
...
+    /**
+      * Prints Spark physical plan.
+      */
+    def printPhysicalPlan: Unit = {
+      println("Spark plan:")
+      implicit val sc = df.sparkSession.sparkContext
+      val sparkPlan: SparkPlan = df.queryExecution.executedPlan
+      // Remove cached inputs from plan
+      val planWithoutCached = sparkPlan.transformDown {
+        case imts: InMemoryTableScanExec => CachedOperatorInput(imts.relation.tableName)
```
Comment (s1ck) [inline spark-cypher/src/main/scala/org/opencypher/spark/impl/DataFrameOps.scala]: Check if IMTS is cached

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/BinaryOperators.scala [hunk 2]
```diff
@@ -54,7 +54,12 @@ private[spark] abstract class BinaryPhysicalOperator extends CAPSPhysicalOperato
 
   def rhs: CAPSPhysicalOperator
 
-  override def execute(implicit context: CAPSRuntimeContext): CAPSPhysicalResult = executeBinary(lhs.execute, rhs.execute)
+  override def execute(implicit context: CAPSRuntimeContext): CAPSPhysicalResult = {
+    val lhsResult = lhs.execute
+    val rhsResult = rhs.execute
+    maybeChildResults = Some(List(lhs -> lhsResult, rhs -> rhsResult))
```
Comment (s1ck) [inline spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/BinaryOperators.scala]: (as discussed) remove pls. trade off not worth the mutable state

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/NAryOperators.scala [hunk 3]
```diff
@@ -28,15 +28,28 @@ package org.opencypher.spark.impl.physical.operators
 
 import org.opencypher.okapi.api.graph.QualifiedGraphName
 import org.opencypher.spark.impl.physical.{CAPSPhysicalResult, CAPSRuntimeContext}
-import org.opencypher.spark.impl.{CAPSGraph, CAPSRecords, CAPSUnionGraph}
 import org.opencypher.spark.impl.util.TagSupport._
+import org.opencypher.spark.impl.{CAPSRecords, CAPSUnionGraph}
+
+
+private[spark] abstract class NAryPhysicalOprator extends CAPSPhysicalOperator {
```
Comment (s1ck) [inline spark-cypher/src/main/scala/org/opencypher/spark/impl/physical/operators/NAryOperators.scala]: typo

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/DataFrameOps.scala (hunk 1 above)
Comment (pstutz) [reply spark-cypher/src/main/scala/org/opencypher/spark/impl/DataFrameOps.scala]: Improved the dummy operator name and made the replacement feature configurable. I guess the really interesting thing to know is if that table has been both cached *and* computed already, but I'm not sure how to figure out the latter.

**Pull request merged**